csharp/src/Apache.Arrow/Arrays/ArrayData.cs

```diff
@@ -73,5 +74,18 @@ public void Dispose()
                 }
             }
         }
+
+        public ArrayData Slice(int offset, int length)
+        {
+            if (offset > Length)
+            {
+                throw new ArgumentException($"Offset {offset} cannot be greater than Length {Length} for Array.Slice");
+            }
+
+            length = Math.Min(Length - offset, length);
+            offset += Offset;
+
+            return new ArrayData(DataType, length, NullCount, offset, Buffers, Children);
```

Comment:
Is reusing the same `NullCount` really correct here? What if we sliced some of the `null`s away?

Reply:
Yup, I was going to pose that exact question before merging the PR. Would folks be ok with setting the NullCount here to -1 to signify that we don't know the NullCount yet? NullCount can then be made into a property that is evaluated when needed. The C++ lib does this at the moment. If that idea is acceptable, I don't want to do that in this PR though, I'd rather open a separate issue for.

Reply:
I think that sounds like a fine plan - both setting a sentinel value to mean "please compute when needed", and to log a separate JIRA issue for this.